client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/multimap/metadata/MetadataCollectionImpl.java

```diff
@@ -0,0 +1,71 @@
+package org.infinispan.client.hotrod.impl.multimap.metadata;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+import org.infinispan.client.hotrod.multimap.MetadataCollection;
+
+public class MetadataCollectionImpl<V> implements MetadataCollection<V> {
+   private final Collection<V> collection;
+   private final long created;
+   private final int lifespan;
+   private final long lastUsed;
+   private final int maxIdle;
+   private final long version;
+
+   public MetadataCollectionImpl(Collection<V> collection, long created, int lifespan, long lastUsed, int maxIdle, long version) {
```

Comment:
Maybe add another constructor that just takes the Collection. Also we should document on this class what time unit these values are assumed to be in. I am guessing MILLISECONDS.

Reply:
ok